### PR TITLE
AngularDeprecation: Show warnings in panel edit for angular panels

### PR DESCRIPTION
**Pull request by torkelo**
Screenshot from 2023-05-09 07-45-59

### CODE REVIEW

**Comment by tskarhed** [comment]
I would prefer an error to indicate the urgency. A link to a resource that isn't to plugin developer-centric would also be good. Maybe just a link to the plugin's definition, like you mentioned.

**Comment by sympatheticmoose** [comment]
Nice!

> Is this a good placement?

Would this only show if a user edits a panel? Rather than just viewing a dashboard (for example).

This placement could be good for discouraging new panels using angular components. But we'd likely need this in combination with other visual indicators in the catalog and probably dashboards (but for these we could make them dismissible). 


> Alert severity? Warning or Error?

Warning feels appropriate to me.

> Refine text / wording
> We can also include info from plugin definition like links & author name & URL (but would make it a bit larger)
> Or link the the plugin details page (under admin)

Suggested wording: "The selected plugin is using Angular which is deprecated and will stop working in an upcoming release. [Learn more](https://grafana.com/docs/grafana/next/developers/angular_deprecation/).

The docs will have a list of all angular plugins and potential alternatives where available; the catalog will also display which are Angular based.

**Comment by torkelo** [comment]
Updated wording 
Screenshot from 2023-05-09 07-45-59

File: public/app/features/dashboard/components/PanelEditor/AngularPanelPluginWarning.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+          <p>The selected visualization plugin is using deprecated plugin APIs.</p>
+          <ul>
+            <li>
+              <a
+                href="https://grafana.com/docs/grafana/latest/developers/angular_deprecation/"
+                className="external-link"
+                target="_blank"
+                rel="noreferrer"
+              >
+                Read more on angular deprecation
```
Comment (dprokop) [inline public/app/features/dashboard/components/PanelEditor/AngularPanelPluginWarning.tsx:27]: Think Angular (as a proper name) is written like this:
```suggestion
                Read more on Angular deprecation
```

**Review by dprokop — APPROVED** [review]

File: public/app/features/dashboard/components/PanelEditor/AngularPanelPluginWarning.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+export function AngularPanelPluginWarning({ plugin }: Props) {
+  const styles = useStyles2(getStyles);
+
+  return (
+    <div className={styles.wrapper}>
+      <Alert title="Angular panel plugin" severity="warning">
+        <div className="markdown-html">
+          <p>The selected visualization plugin is using deprecated plugin APIs.</p>
```
Comment (torkelo) [inline public/app/features/dashboard/components/PanelEditor/AngularPanelPluginWarning.tsx:18]:
```suggestion
          <p>The selected panel plugin is using deprecated plugin APIs.</p>
```

**Review by juanicabanas — APPROVED** [review]

**Pull request merged**